keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,319 @@
+
+All beta PDB endpoints are covered by e2e tests: https://apisnoop.cncf.io/1.20.0/beta/policy
+
+#### Conformance tests
+
+The following e2e tests will be included in the conformance tests:
+
+ * DisruptionController
+   * should block an eviction until the PDB is updated to allow it
+   * create a PodDisruptionBudget
+   * delete collection of PodDisruptionBudget
+   * delete a PodDisruptionBudget
+   * list or watch objects of kind PodDisruptionBudget
+   * partially update the specified PodDisruptionBudget
+   * partially update status of the specified PodDisruptionBudget
+   * read the specified PodDisruptionBudget
+   * read status of the specified PodDisruptionBudget
+   * replace the specified PodDisruptionBudget
+   * replace status of the specified PodDisruptionBudget
+
+### Graduation Criteria
+
+Graduation to GA:
+- [x] Implement Mutable PDBs
+- [ ] Address performance issues
+- [ ] Pass conformance tests
+- [ ] Update documents to reflect the changes
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodDisruptionBudget
+    - Components depending on the feature gate: kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  No
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes
+
+* **What happens if we reenable the feature if it was previously rolled back?** 
+  No changes expected
+
+* **Are there any tests for feature enablement/disablement?**
+  Not applicable, given we're missing framework allowing switching feature-gates during e2e.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  - If the leader election for the controller-manager fails, we can end up with two instances which will fight over the same PDB resources
+  - A panic in the disruption controller will crash the controller-manager
+
+* **What specific metrics should inform a rollback?**
+  - Unexpected controller-manager crashes
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Manual test is planned once the implementation is finished.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  No
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  - Check number of PDBs in cluster with `etcd_object_counts{resource="poddisruptionbudgets.policy"}`
+  - Queue-related metrics to make sure the controller is functioning.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [x] Metrics
+    - Metric name: `workqueue_depth`
```

Comment:
Can you please provide the exact metrics to ensure we only catch the disruption-controller ones?

Reply:
Updated the metrics here to include the necessary label for the disruption controller. I got the names here from https://github.com/kubernetes/kubernetes/blob/506156842cde2180ce5351ca298a13c59f191d89/staging/src/k8s.io/component-base/metrics/prometheus/workqueue/metrics.go#L30-L37